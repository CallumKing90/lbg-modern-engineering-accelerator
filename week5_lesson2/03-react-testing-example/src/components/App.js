import React from 'react';
import Navbar from './Navbar';
import Main from './Main';
import '../styles/style.scss';

const App = () => {
  const [page, setPage] = React.useState('Click a button!');
  const onNavChange = (pageToGoTo) => setPage(pageToGoTo);

  return (
    <>
      <h1>Learn React Testing</h1>
      <Navbar onNavChange={onNavChange} />
      <Main page={page} />
    </>
  );
};

export default App;
