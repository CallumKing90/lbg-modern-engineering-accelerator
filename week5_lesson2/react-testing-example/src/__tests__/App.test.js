import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import App from '../components/App';

describe('<App />', () => {
  it('renders the welcome message', () => {
    render(<App />);
    expect(screen.getByText('Learn React Testing')).toBeInTheDocument();
  });

  it('matches snaphot on initial render', () => {
    const component = render(<App />);
    expect(component.baseElement).toMatchSnapshot();
  });

  it('matches snapshot after clicking About', () => {
    const component = render(<App />);

    // lets interact with the page
    const aboutButton = screen.getByText('About');
    fireEvent.click(aboutButton);

    expect(component.baseElement).toMatchSnapshot();
  });
});
