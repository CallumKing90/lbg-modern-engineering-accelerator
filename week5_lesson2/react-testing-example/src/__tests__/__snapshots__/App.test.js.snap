// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> matches snaphot on initial render 1`] = `
<body>
  <div>
    <h1>
      Learn React Testing
    </h1>
    <nav>
      <ul>
        <li>
          <button>
            Home
          </button>
        </li>
        <li>
          <button>
            About
          </button>
        </li>
        <li>
          <button>
            Contact
          </button>
        </li>
        <li>
          <button>
            Star Wars
          </button>
        </li>
      </ul>
    </nav>
    <main>
      <p
        class="large"
      >
        Click a button!
      </p>
    </main>
  </div>
</body>
`;

exports[`<App /> matches snapshot after clicking About 1`] = `
<body>
  <div>
    <h1>
      Learn React Testing
    </h1>
    <nav>
      <ul>
        <li>
          <button>
            Home
          </button>
        </li>
        <li>
          <button>
            About
          </button>
        </li>
        <li>
          <button>
            Contact
          </button>
        </li>
        <li>
          <button>
            Star Wars
          </button>
        </li>
      </ul>
    </nav>
    <main>
      <p
        class="large"
      >
        About
      </p>
    </main>
  </div>
</body>
`;
